src/compiler/utilities.ts

```diff
@@ -3990,6 +3990,13 @@ namespace ts {
         return isPropertyAccessExpression(node) && isEntityNameExpression(node.expression);
     }
 
+    export function isConstructorAccessExpression(expr: Expression): expr is AccessExpression {
+        return (
+            isPropertyAccessEntityNameExpression(expr) && idText(expr.name) === "constructor" ||

```

Comment:
I'd loosen this - there's no need for whatever's left of the `constructor` to be so strict, so long as the references match. (Plus, as written this prevents something like `this[mySymbol].constructor === Cls` from working)
```suggestion
            isPropertyAccessExpression(expr) && idText(expr.name) === "constructor" ||
```
especially since the `ElementAcessExpression` case below is already more general and allows arbitrary LHS expressions.

Reply:
Ah good point, thanks. I'll change this.